app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/tagsselector/TagsSelectorPresenter.kt

```diff
@@ -3,95 +3,125 @@ package space.taran.arknavigator.mvp.presenter.adapter.tagsselector
 import android.util.Log
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
+import space.taran.arknavigator.R
+import space.taran.arknavigator.mvp.model.UserPreferences
 import space.taran.arknavigator.mvp.model.repo.index.ResourceId
+import space.taran.arknavigator.mvp.model.repo.index.ResourceKind
 import space.taran.arknavigator.mvp.model.repo.index.ResourcesIndex
 import space.taran.arknavigator.mvp.model.repo.tags.TagsStorage
 import space.taran.arknavigator.mvp.view.ResourcesView
+import space.taran.arknavigator.ui.resource.StringProvider
 import space.taran.arknavigator.utils.Popularity
 import space.taran.arknavigator.utils.TAGS_SELECTOR
 import space.taran.arknavigator.utils.Tag
-import space.taran.arknavigator.utils.Tags
 import java.nio.file.Path
+import javax.inject.Inject
 import kotlin.reflect.KSuspendFunction1
 
+sealed class TagItem {
+    data class DefaultTagItem(val tag: Tag) : TagItem()
+    data class KindTagItem(val kind: ResourceKind) : TagItem()
+}
+
 class TagsSelectorPresenter(
     private val viewState: ResourcesView,
     private val prefix: Path?,
     private val scope: CoroutineScope,
     private val onSelectionChangeListener: KSuspendFunction1<Set<ResourceId>, Unit>
 ) {
+    @Inject
+    lateinit var stringProvider: StringProvider
```

Comment:
Is it possible to inject just `Context`? So it would be possible to remove code duplicate below (initializing `kindToString`). We have the same code in `TagsSelectorAdapter` which might be forgotten to be updated when we add more resource kinds. Ideally, this map creation should belong to `ResourceMeta` layer.

Reply:
Extracted `kindToString` to `StringProvider`. We do not want to inject the `Сontext` into the business logic, because then it will be impossible to use unit tests, since we will depend on the android. It is better to have pure Kotlin and use abstractions that we can replace in tests.